cmake/external/onnxruntime_external_deps.cmake

```diff
@@ -542,7 +542,9 @@ if(TARGET ONNX::onnx_proto AND NOT TARGET onnx_proto)
   add_library(onnx_proto ALIAS ONNX::onnx_proto)
 endif()
 
-find_package(Eigen3 CONFIG)
+if (onnxruntime_USE_PREINSTALLED_EIGEN)
```

Comment:
onnxruntime_USE_PREINSTALLED_EIGEN is not used for indicating whether find_package should be used for looking for Eigen or not.  It is a deprecated feature. I even couldn't remember why it was added. It is used with another cmake variable. 

By default, cmake always call find_package first, unless FETCHCONTENT_TRY_FIND_PACKAGE_MODE is set to NEVER.